From f0cc12771e61bbc78348f4df7ba39bf3e4da9903 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Rafa=C5=82=20Dzi=C4=99giel?= <rafostar.github@gmail.com>
Date: Mon, 13 Sep 2021 09:47:29 +0200
Subject: [PATCH] va: h264dec: Try to use ConstrainedBaseline or Main to decode BaseLine

A simplified verson of the orginal patch until compilance prop is merged into GStreamer.
For more info and full patch see https://gitlab.freedesktop.org/gstreamer/gst-plugins-bad/-/merge_requests/2428
---
 sys/va/gstvah264dec.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/sys/va/gstvah264dec.c b/sys/va/gstvah264dec.c
index 571eddf7a..0037c5bf6 100644
--- a/sys/va/gstvah264dec.c
+++ b/sys/va/gstvah264dec.c
@@ -601,12 +601,8 @@ _get_profile (GstVaH264Dec * self, const GstH264SPS * sps, gint max_dpb_size)
 
   switch (sps->profile_idc) {
     case GST_H264_PROFILE_BASELINE:
-      /* A.2 compliant */
-      if (sps->constraint_set0_flag || sps->constraint_set1_flag
-          || sps->constraint_set2_flag) {
-        profiles[i++] = VAProfileH264ConstrainedBaseline;
-        profiles[i++] = VAProfileH264Main;
-      }
+      profiles[i++] = VAProfileH264ConstrainedBaseline;
+      profiles[i++] = VAProfileH264Main;
       break;
     case GST_H264_PROFILE_EXTENDED:
       if (sps->constraint_set1_flag) {  /* A.2.2 (main profile) */
-- 
2.26.3

